README.md

```diff
@@ -365,6 +365,52 @@ Returns: `URL`
 * **protocol** `string` (optional)
 * **search** `string` (optional)
 
+## DNS caching
+
+Undici provides DNS caching via the `DNSResolver` class.
+
+This functionality is coming from [cacheable-lookup](https://github.com/szmarczak/cacheable-lookup/tree/9e60c9f6e74a003692aec68f3ddad93afe613b8f) package.
+
+By default a global DNSResolver will be created and passed down by the agent to the pool.
+Here is an example of how to configure a custom DNSResolver:
+
+```js
+new Agent({
+    dnsResolver: {
+      lookupOptions: { family: 6, hints: ALL }
+    }
+  })
+```
+
+To disable DNS caching for an agent:
+
+```js
+new Agent({
+  dnsResolver: {
+    disable: true
+  }
+})
+```
+
+Provide your custom DNS resolver, it must implement the method `lookup`:
+
+```js
+new Agent({
+  DNSResolver: new MyCustomDNSResolver()
+})
+```
+
+All arguments match [`cacheable-lookup`](https://github.com/szmarczak/cacheable-lookup/tree/9e60c9f6e74a003692aec68f3ddad93afe613b8f) package
+
+Extra arguments available in Undici:
+
+* **lookupOptions**
+  * **family** `4 | 6 | 0` - Default: `0`
+  * **hints** [`getaddrinfo flags`](https://nodejs.org/api/dns.html#supported-getaddrinfo-flags)
+  * **all** `Boolean` - Default: `false`
+
+* **roundRobinStrategy** `'first' | 'random'` - Default: `'first'`
```

Comment:
this should be called `scheduling`. What does `first` vs `random` do?

Reply:
done